src/shared/utilities/timeoutUtils.ts

```diff
@@ -16,16 +18,18 @@ export class Timeout {
     private readonly timeoutLength: number
     private readonly timerPromise: Promise<void>
     private timerTimeout?: NodeJS.Timeout
-    private timerResolve?: (value?: void | PromiseLike<void> | undefined) => void
+    private timerResolve?: (value?: void | PromiseLike<undefined> | undefined) => void
+    private timerReject?: (value?: Error | PromiseLike<Error> | undefined) => void
     public constructor(timeoutLength: number) {
         this.startTime = Date.now()
         this.originalStartTime = this.startTime
         this.endTime = this.startTime + timeoutLength
         this.timeoutLength = timeoutLength
         this.timerPromise = new Promise<void>((resolve, reject) => {
-            this.timerTimeout = setTimeout(() => reject(new Error(TIMEOUT_ERROR_MESSAGE)), timeoutLength)
+            this.timerReject = reject
             this.timerResolve = resolve
         })
+        this.timerTimeout = setTimeout(() => this.timerReject!(new Error(TIMEOUT_EXPIRED_MESSAGE)), timeoutLength)
```

Comment:
Nit: `this.timerReject!(new Error(TIMEOUT_EXPIRED_MESSAGE)` -> `killTimer(true)`

Reply:
`killTimer(true)` is actually for cancelling Timeouts, the error here is for an expired Timeout. Need a way to distinguish between the two.